fem/coefficient.hpp

```diff
@@ -68,6 +70,18 @@ class Coefficient
       return Eval(T, ip);
    }
 
+   /** @brief Evaluate the coefficient at all the quadrature points given by the
+       IntegrationRule @a ir on the FiniteElementSpace @a fes. */
+   virtual void Eval(const FiniteElementSpace &fes, const IntegrationRule &ir,
+                     Vector &qcoeff);
```

Comment:
For mixed meshes, we'll need to use different `IntegrationRule`s for different elements. The class `QuadratureSpace` is probably a better choice from that point of view.

Also, for a general coefficient, there is no need to provide a `FiniteElementSpace`, just the `Mesh` is sufficient. If we use `QuadratureSpace` then we do not need an extra argument -- `QuadratureSpace` already has a pointer to the `Mesh` as part of its definition.

Reply:
I tried using only the `Mesh` class but it doesn't work, I'm using some functions that are only in the `FiniteElementSpace` class. (Same for the `QuadratureSpace` class)

Reply:
Can you clarify: which `FiniteElementSpace` methods are you using and where?

Reply:
`GetTraceElement` in `coefficient.cpp:51`.